code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -252,6 +256,25 @@ Please contact me on #coderbus IRC. ~Carn x
 		list_huds = hud_list.Copy()
 		list_huds += backplane // Required to mask HUDs in context menus: http://www.byond.com/forum/?post=2336679
 
+//TYPING INDICATOR CODE.
+	if(client && !stat) //They have a client & aren't dead/KO'd? Continue on!
+		if(typing_indicator && hud_typing) //They already have the indicator and are still typing
+			overlays += typing_indicator //This might not be needed? It works, so I'm leaving it.
+			list_huds += typing_indicator
+			typing_indicator.invisibility = invisibility
+
+		else if(!typing_indicator && hud_typing) //Are they in their body, NOT dead, have hud_typing, do NOT have a typing indicator. and have it enabled?
+			typing_indicator = new
+			typing_indicator.icon = 'icons/mob/talk.dmi'
+			typing_indicator.icon_state = "[speech_bubble_appearance()]_typing"

```

Comment:
This should be in a single line to avoid appearance churn (idk if this is an image or icon but both the image() and icon() procs support passing an icon and state)

Reply:
Ok, it's NEITHER, it's an OBJECT used as an overlay. Okay then.